### ISSUE TITLE
Does not seem to work on chrome android ?

### ISSUE THREAD

**Issue by plagasul** [mnater/Hyphenopoly#78]
May it be hyphenopoloy is NOT working on chrome android? Did I miss anything?

I am developing a site on localhost and while on desktop chrome devtools, mobile mode, it works, if I visit it from my cellphone on the local network, default android chrome browser, I don't see any hyphenation.

![image](https://user-images.githubusercontent.com/13966673/57588228-615eb000-7511-11e9-9795-1837bdb7991d.png)
![image](https://user-images.githubusercontent.com/13966673/57588234-79ceca80-7511-11e9-92ca-6f46e6b2a2c0.png)

Thanks

**Comment by mnater** [comment]
Hi

I don't know. I don't have a real android device at hand to test. It works in my android emulator, though.

Could you please send the complete code or a link, where I can test (including exact Android Version and Device).

Chrome on Android has native CSS-hyphens support (like Chrome on Mac, but not Chrome on Windows). Did you enable CSS-hyphenation or use "FORCEHYPHENOPOLY"?

**Comment by plagasul** [comment]
Hello, thanks for your answer.

Here is a link that should show a similar layout (different branch) and no hyphens on android : https://vesselartproject.org/final

I am on a Huawei P Smart 2019, with Chrome 74.0.3729.112, and Android 9 and EMUI 9.0.1

The CSS code for that element is:
```
.home-grid-container {
  align-self: flex-start;
  display: grid;
  grid-template-columns: min-content 2fr 1fr;
  grid-gap: 0px;
  /* width: 97%; */
  overflow-y: auto;
  overflow-x: hidden;
  font-size: 12vh;
  line-height: normal;
  /* hyphenopoly test */
  hyphens: auto;
  -ms-hyphens: auto;
  -moz-hyphens: auto;
  -webkit-hyphens: auto;
}
```

Hyphenopoly is initialized on head tag using this code (paths and last line use a CMS specific php helper):

```
<script>
var Hyphenopoly = {
		require: {
				//"it": "hippopotomonstrosesquippedaliofobia",
				"en-us": "Supercalifragilisticexpialidocious"
		},
		setup: {
				selectors: {
					".home-grid-container": {},
					".grid-container": {},
				}
		}
		,
		paths: {
			// Use kirby to output here the right route
			patterndir: "<?= $kirby->url('assets') ?>/js/hyphenopoly/patterns/", //path to the directory of pattern files
			maindir: "<?= $kirby->url('assets') ?>/js/hyphenopoly/" //path to the directory where the other ressources are stored
		}
};
</script>
<?= js('assets/js/hyphenopoly/Hyphenopoly_Loader.js') ?>
```

Thank you

**Comment by mnater** [comment]
Thanks for the information.

I'm able to reproduce the issue on the phone emulator in android studio. It's a strange case and I don't have a soultion, yet.

1. I've noticed that there are no `lang`-attributes in your html. The hyphenation algorithm (be it native CSS or Hyphenopoly) needs to no what language the text is written. So make sure to add proper language tags. – But that didn't change anything.

2. Chrome (Desktop) recognises the emulator as a remote device. So we are able to remotely debug the page as it is run on android. If I comment out [line #772 in Hyphenopoly_Loader.js](https://github.com/mnater/Hyphenopoly/blob/master/Hyphenopoly_Loader.js#L772) the Hyphenopoly object persist and we can look at it in Chromes DevTools console:
````javascript
clientFeat: {
    langs: {en-us: "CSS"}
    polyfill: false
    wasm: true
}
````
In other words: the feature tests run by Hyphenopoly_Loader.js tell us that Chrome on Android supports CSS-hyphenation for `en-us` and WebAssembly and therefore polyfilling isn't necessary.
I double-checked and Chrome on Android really supports CSS-hyphenation.
Also, if we bypass feature checking and enforce the usage of Hyphenopoly (["FORCEHYPHENOPOLY"](https://github.com/mnater/Hyphenopoly/wiki/Global-Hyphenopoly-Object#require)) the text is correctly hyphenated (given you've set the `lang`-attribute, see pt 1). So Hyphenopoly works correctly and this seems to be an issue with Chrome.

Something seems to prevent CSS-hyphenation to work in your case. I'll try to narrow down what this is…

**Comment by mnater** [comment]
> Something seems to prevent CSS-hyphenation to work in your case. I'll try to narrow down what this is…

I was wrong about this. Hyphenation works for english but Android doesn't support hyphenation for italian. So the text "L'Istituto per l'Immaginazione del Mediterraneo" isn't hyphenated (and expands the grid-track so that the text in the lines above don't need to be hyphenated).

In your html: set the language tags correctly. For example set `lang="en-us"` in the `<html>` tag and then set `lang="it"` for each element in italian.

For Hyphenopoly: `require` both (`en-us` and `it`) Hyphenopoly will then detect if the browser supports CSS-Hyphenation.

I'll close this issue for now. Feel free to reopen if I was wrong🤪

**Comment by jallusparta** [comment]
@plagasul did you get this to work in android chrome by requiring italian in hyphenpoly?

My problem is, that as android doesn't have hyphens dictionary I need either, hyphens doesn't work on chrome but does in other browsers. Can I bypass this problem by using hyphenpoly and importing the needed hyphens dictionary?

**Comment by plagasul** [comment]
Apologies @jallusparta , it is as @mnater explained. Setting the language attribute in the html element and requiring italian when instantiating hyphenopoly works.
